docker/Dockerfile

```diff
@@ -69,6 +69,16 @@ RUN set -e; \
         esac)"; \
     . <(sed 's/^\(..*\)/export DISTRO_\1/' '/etc/os-release'); \
     case "$DISTRO_ID-$DISTRO_VERSION_ID" in \
+    'debian-12') \
+        export DEB_SRC_LIST=/etc/apt/sources.list; \
+        [ -e "$DEB_SRC_LIST" ] || \
```

Comment:
@xkszltl Do you think this is a reasonable workaround?

Reply:
So where's the official one, they won't just disappear right? Are they moved to .list.d? If so your lines may still be shadowed and we shouldn't duplicate the repo.

Reply:
And add-apt-repository claimed to support .list.d, so you'd better figure out what's going on.

Reply:
They ARE under sources.list.d, while the format also changed. That is probably why 
`add-apt-repository` does not work.

```
# cat /etc/apt/sources.list.d/debian.sources 
Types: deb
# http://snapshot.debian.org/archive/debian/20230612T000000Z
URIs: http://deb.debian.org/debian
Suites: bookworm bookworm-updates
Components: main
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg

Types: deb
# http://snapshot.debian.org/archive/debian-security/20230612T000000Z
URIs: http://deb.debian.org/debian-security
Suites: bookworm-security
Components: main
Signed-By: /usr/share/keyrings/debian-archive-keyring.gpg
```

Reply:
We can always switch to the new tooling if there are any. Like I said, we just use it as a workaround to unblock.